cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -106,7 +106,10 @@ size_type sort_groupby_helper::num_keys(cudaStream_t stream) {
 
 column_view sort_groupby_helper::key_sort_order(cudaStream_t stream) {
   if (_key_sorted_order)
-    return _key_sorted_order->view();
+    return cudf::experimental::detail::slice(
+      _key_sorted_order->view(),
+      0,
+      num_keys(stream));
```

Comment:
This isn't enough. This will not return sliced order when called the first time. Only at subsequent calls.

Reply:
I'm unable to add review comment on that line but check the end of this function. You need to add this over there as well.

Reply:
Gotcha

Reply:
Do we then need to slice here at all?

Reply:
Ah I guess we do.

Reply:
Fixed - there were two places where we were returning. Please have a look to make sure I'm not doing something wrong. Thanks!